### ISSUE TITLE
out-of-the-box unexpected token with create-react-app

### ISSUE THREAD

**Issue by afontcu** [orta/vscode-jest#125]
Hi there,

### Environment

1. `node -v`:  v6.10.3
2. `npm -v`: 4.6.1
4. `npm ls react-scripts` (if you haven’t ejected): 1.0.10

1. Operating system: MacOS

### Steps to Reproduce

1. install `create-react-app`
2. install `jest` extension for Visual Studio Code (the [recommended one](https://github.com/orta/vscode-jest))
3. reload it and try to debug

### Expected Behavior

Pass the tests (as it does in CLI)

### Actual Behavior

```
 FAIL  src/App.test.js

  ● Test suite failed to run

    .../src/App.test.js: Unexpected token (7:18)
        5 | it('renders without crashing', () => {
        6 |   const div = document.createElement('div');
      > 7 |   ReactDOM.render(<App />, div);
          |                   ^
        8 | });
        9 | 
```

Thanks!

**Comment by orta** [comment]
This is working for me on my default create-react-app example, https://github.com/orta/vscode-jest/tree/master/integrations/create-react-example - which I just updated now to the latest version

![screen shot 2017-07-02 at 06 59 04](https://user-images.githubusercontent.com/49038/27769246-fc0ac03e-5ef3-11e7-989a-d83b33648014.png)

Any chance you can run the extension in debug mode and have a look at what causing it?

**Comment by afontcu** [comment]
Hi there,

I'm sorry, it worked flawlessly when I tried it in another project. Then I downloaded CRA again and it worked as well, so I'm assuming there was something weird in my previous installation/configuration, because it is still not working there. 

Thanks for your time.

**Comment by Cerberus** [comment]
I got the same error. In my case, the cause was after I installed `jest`, I've updated `package.json` from
`"test": "react-scripts test --env=jsdom",`
to this
`"test": "jest"`
ref: react-scripts v. 1.0.7

**Comment by seanpoulter** [comment]
@orta, can I help get this one fixed? Any tips on what I'm looking for if I get the extension running in debug mode?

It's failing from a clean create-react-app project with react-scripts@1.0.14, jest@20.0.4, vscode-jest v2.4.2.

**Comment by orta** [comment]
Honestly, I'm not that sure, maybe that the Jest config isn't being respected somehow?

**Comment by seanpoulter** [comment]
You're right @orta. The `defaultCreateReactPathWindows` isn't valid for the current version of react-scripts. While I get VS Code loading the cloned extension to start debugging:
* Did you have a strategy in mind for supporting different versions of react-scripts?
* Should we extract the detection from `src/helpers.ts` to handle it?
* Would you like me to go through the issues to help indicate which ones are trying to get create-react-app working?

**Comment by orta** [comment]
For Jest, the extension aims to support `major version - 1`- and doesn't strive to break backwards compatibility unless forced to (as we have the jest-extension-support lib _inside jest, so sometimes it does). 

I'm happy to have a similar aim with CRA, I think it's probably a bit easier to handle as it could be a matter of checking for the right files in the right places.

I'm happy for you to do that yep - I generally run projects [like this](https://github.com/Moya/contributors) so I'll give you access rights to handle issues however you like

**Comment by djErock** [comment]
So what is the fix here?

**Comment by seanpoulter** [comment]
You've reproduced the issue today @djErock? I can take a look on my next break.

**Comment by seanpoulter** [comment]
If you're having problems @djErock could you please create a new issue with lots of detail so we can investigate. I've just confirmed create-react-app has worked OK on OS X, Linux, and Windows systems.

If you're asking to summarize the issue, updating the extension should resolve it. There are a couple fixes at play that improved detection of create-react-app, and spawning the runner in a shell on Windows.

**Comment by nhooey** [comment]
I'm getting the same issue with failing tests after creating a project with `create-react-app` and running tests with `jest`:

# Environment

1. `node -v`: `v9.8.0`
2. `npm -v`: `5.6.0`
3. `npm ls react-scripts`:
```
react-test@0.1.0 /private/tmp/react/react-test
└── react-scripts@1.1.1
```

# Expected Behaviour

Pass the tests

# Actual Behaviour

I am currently getting `Unexpected token` when running `react test` after installing `jest` and replacing the `package.json` `test` line with `jest`.

# Steps to Reproduce

Here's what happens when I start from scratch:

**Install and set up `nodeenv`:**
```bash
$ mkdir /tmp/react
$ cd /tmp/react
$ brew install nodeenv
    (installs nodeenv)
$ nodeenv .venv
 * Install prebuilt node (9.8.0) ..... done.
$ source .venv/bin/activate
```

**Create `react` app:**
```bash
(.venv) $ npx create-react-app react-test
npx: installed 67 in 5.669s
[...snipped...]
> fsevents@1.1.3 install /private/tmp/react/react-test/node_modules/fsevents
> node install
[fsevents] Success: "/private/tmp/react/react-test/node_modules/fsevents/lib/binding/Release/node-v59-darwin-x64/fse.node" is installed via remote
> uglifyjs-webpack-plugin@0.4.6 postinstall /private/tmp/react/react-test/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js
+ react-scripts@1.1.1
+ react@16.2.0
+ react-dom@16.2.0
added 1440 packages in 59.115s
[...snipped...]
```

**Run vanilla react tests:**
```bash
(.venv) $ cd react-test
(.venv) $ npm test
> react-test@0.1.0 test /private/tmp/react/react-test
> react-scripts test --env=jsdom
 PASS  src/App.test.js
  ✓ renders without crashing (17ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        0.925s
Ran all test suites related to changed files.
```

**Install `jest`:**
```bash
(.venv) $ npm install --save jest
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
+ jest@22.4.2
added 112 packages, removed 1 package, updated 35 packages and moved 9 packages in 18.697s
```

**Run tests after installing `jest`:**
```bash
(.venv) $ npm test
 FAIL  src/App.test.js
  ● Test suite failed to run

    TypeError: environment.dispose is not a function

      at Promise.resolve.then (node_modules/react-scripts/node_modules/jest/node_modules/jest-cli/build/runTest.js:102:17)
          at <anonymous>

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.791s, estimated 1s
Ran all test suites related to changed files.
```

**`packge.json` before updated with `jest` test directive:**
```json
{
  "name": "react-test",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "jest": "^22.4.2",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-scripts": "1.1.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}

**Replace `test` directive with `jest`:**
```bash
(.venv) $ sed -i -e 's/"test": .*/"test": "jest",/' package.json
```

**Run `npm test` with `jest`:**
```bash
(.venv) $ npm test

> react-test@0.1.0 test /private/tmp/react/react-test
> jest

 FAIL  src/App.test.js
  ● Test suite failed to run

    /private/tmp/react/react-test/src/App.test.js: Unexpected token (7:18)
         5 | it('renders without crashing', () => {
         6 |   const div = document.createElement('div');
      >  7 |   ReactDOM.render(<App />, div);
           |                   ^
         8 |   ReactDOM.unmountComponentAtNode(div);
         9 | });
        10 |

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.509s
Ran all test suites.
npm ERR! Test failed.  See above for more details.
```

**Comment by nhooey** [comment]
@orta @seanpoulter: This issue still persists as of today with the latest versions.

**Comment by orta** [comment]
I'm a little confused, I don't think this has anything to do with the vscode-jest extension?

**Comment by seanpoulter** [comment]
I'll follow your setup late tonight @nhooey on macOS and Windows. Let's get'r done.

**Comment by seanpoulter** [comment]
**TL;DR**: Nope nope nope. You can't install `jest` when bootstrapped with `create-react-app`.

<br/>
There's a fundamental problem with your expectation to install and run `jest` in a project that has been bootstrapped with `create-react-app`. When using `create-react-app`, you're asking `react-scripts` to do the configuration for you. That means this is an expected error:

```
$ yarn jest
yarn run v1.5.1
$ /vscode-jest/issues/125/app/node_modules/.bin/jest
 FAIL  src/App.test.js
  ● Test suite failed to run

    TypeError: environment.setup is not a function
      
      at node_modules/jest-runner/build/run_test.js:71:23

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.612s, estimated 1s
Ran all test suites.
```

The working assumption is that you're going to continue to use `react-scripts` to interact with the project and let it configure everything. By installing `jest` you're breaking this assumption and all the hard work that `create-react-app` is doing for you. From what I understand, installing and updating `jest` but not the other dependencies is what causes the error above. In my case, I've installed `jest@23` but continue to have `jest-environment-node@20.0.3` installed.

Installing `jest` with `create-react-app` is not supported. Here's an authoritative comment on an Issue stating this: https://github.com/facebook/jest/issues/5119#issuecomment-356120965

**Comment by seanpoulter** [comment]
If you continue to experience problems please create a new issue and reference #125. I've locked the conversation for this issue since we've resolve the original and subsequent issues. 🔒
